x-eithert/src/X/Control/Monad/Trans/Either.hs

```diff
@@ -59,6 +61,10 @@ eitherT f g m =
   either f g =<< runEitherT m
 {-# INLINE eitherT #-}
 
+tryEitherT :: (MonadCatch m, Exception e) => (e -> x) -> m a -> EitherT x m a
```

Comment:
yeah I think @nhibberd would get what he wants by calling `tryEitherT Disaster` so I think it's good as is